### PR TITLE
[segment-api] Added segment-api typings

### PR DESCRIPTION
**Pull request by padcom**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by padcom** [comment]
@RyanCavanaugh the @typescript-bot seems to not update the comment that is supposed to be "live"... How often do the comments get updated?

File: types/segment-api/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Type definitions for segment-api 4.13
+// Project: https://github.com/padcom/segment-api
+// Definitions by: Matthias Hryniszak <https://github.com/padcom>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.0
+
+export interface Traits {
```
Comment (martin-badin) [inline types/segment-api/index.d.ts:7]: Please specify traits https://segment.com/docs/connections/spec/identify/#traits

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/segment-api/index.d.ts (hunk 1 above)
Comment (padcom) [reply types/segment-api/index.d.ts:25]: Done

**Review by martin-badin — APPROVED** [review]
Awesome. Looks good to me

thanks @padcom

**Comment by padcom** [comment]
Thank you!

**Comment by DanielRosenwasser** [comment]
Thank *you* for the PR and the patience.

**Pull request merged**